File: dev-docs/snippets/databases/execute-batch.swift
import JsBaoClient

// Execute a batch of records using a named mutation operation.
// Each batch element wraps one record's params.
func executeBatch(client: JsBaoClient, databaseId: String, operationName: String) async throws {
  // #region example
  let result = try await client.databases.executeBatch(
    databaseId: databaseId,
    operationName: operationName,
    batch: [
      DatabaseBatchOperation(params: ["id": "p1", "name": "Widget"]),
      DatabaseBatchOperation(params: ["id": "p2", "name": "Gadget"]),
    ]
  )
  // #endregion example
  _ = result
}

File: dev-docs/snippets/databases/execute-batch.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Execute a batch of records using a named mutation operation.
export async function executeBatch(
  client: JsBaoClient,
  databaseId: string,
  operationName: string,
) {
  // #region example
  const result = await client.databases.executeBatch(databaseId, operationName, [
    { params: { id: "p1", name: "Widget" } },
    { params: { id: "p2", name: "Gadget" } },
  ]);
  // #endregion example
  return result;
}
